common/queue/src/main/java/org/thingsboard/server/queue/discovery/HashPartitionService.java

```diff
@@ -181,9 +186,20 @@ public void removeQueue(TransportProtos.QueueDeleteMsg queueDeleteMsg) {
     @Override
     public TopicPartitionInfo resolve(ServiceType serviceType, String queueName, TenantId tenantId, EntityId entityId) {
         TenantId isolatedOrSystemTenantId = getIsolatedOrSystemTenantId(serviceType, tenantId);
+        if (queueName == null) {
+            queueName = MAIN_QUEUE_NAME;
+        }
         QueueKey queueKey = new QueueKey(serviceType, queueName, isolatedOrSystemTenantId);
         if (!partitionSizesMap.containsKey(queueKey)) {
-            queueKey = new QueueKey(serviceType, isolatedOrSystemTenantId);
+            if (isolatedOrSystemTenantId.isSysTenantId()) {
+                queueKey = new QueueKey(serviceType, TenantId.SYS_TENANT_ID);
+            } else {
+                queueKey = new QueueKey(serviceType, queueName, TenantId.SYS_TENANT_ID);
```

Comment:
Please add TODO to improve TransportService.getQueueRoutingInfo for cases when it does not fit into the reply (kafka limitations). No need to fix this now. But a TODO is always great :-)